### ISSUE TITLE
Capsule Collider on Pawn does not really work as expected.

### ISSUE THREAD

**Issue by workingmagic** [ruyo/VRM4U#80]
Hello Ruyo!

Actually I am not sure what happened but somehow the capsule collider is not really working as it should. I even tested it with a complete fresh 4.26.2 project. It also does not collide with other pawns. There is some collision with statics but it feels like the capsule collider has a size of 0. Here are some screenshots that show where the collision happens:

![UE4Editor_tI8JfdRpOX](https://user-images.githubusercontent.com/72321604/123797559-95aaab80-d8e6-11eb-9acf-0921ea53a6af.png)
![UE4Editor_QOQfl9Vcy3](https://user-images.githubusercontent.com/72321604/123797414-71e76580-d8e6-11eb-95a8-346c7da37d92.png)
![UE4Editor_qPFvPZbh2T](https://user-images.githubusercontent.com/72321604/123797484-8166ae80-d8e6-11eb-887e-843e628d62e1.png)

**Comment by workingmagic** [comment]
Update: So, I was being able to fix it - but don't know if this is the correct way!

An Capsule collider on VRM4U is in the root, which is correct to get collisions, but having a shape of 1,0 Capsule Half Height and 1,0 Capsule Radius as a base. There is a second capsule inside the mesh but that's not working, no matter which settings.

Question: Is there a special reason for it? I wonder how or if the capsule-half height will be calculated when loading the asset lists or even VRM on runtime.

**Comment by ruyo** [comment]
Hi.
The explanation is complicated, so I'm not sure if I translated it well. Please feel free to point it out.

In the standard UE4 pawn, the origin is at the center of the character (around the waist).
The actor that VRM4U prepares for runtime loading is set to have its origin at the feet.  The reason there are two collisions is to change the origin.

When using the standard pawn, you need to accurately measure the waist coordinates of your character to get it to stand on the ground correctly. You can use this to adjust the size of the capsule collision and the coordinates of the skeletalmesh.

When using BP_VrmCharacterBase, there is no need to adjust the character to stand on the ground.
This is because the origin of the skeletalmesh is in the feet, and the coordinates for grounding are accurate.

![image](https://user-images.githubusercontent.com/6472835/123837104-6ecf9200-d945-11eb-8af3-815dfa475ab1.png)

However, as you pointed out, the collision size remains at the default value as it is.
It is possible to change the collision depending on the character, but getting it to ground and size accurately is a difficult problem...

thanks!

**Comment by ruyo** [comment]
In the next version, I' ll set the collision preset to Pawn.
I think this will make the interaction work as well as the standard Pawn.

![image](https://user-images.githubusercontent.com/6472835/123838796-58c2d100-d947-11eb-87e1-1c7f40624ff5.png)

**Comment by workingmagic** [comment]
Yeah I get it! Well I am not sure why the inherited Capsule does not work then. I've read through a couple of forum threads and it seems that the "root" capsule collider is the only one that works. In VRM4U this one is at a scale 1,1.

Does the inherited capsule collider works for you? Maybe it's some bug or something else that prevents it for me.
